RegisterPayments/handler.ts

```diff
@@ -0,0 +1,391 @@
+                data: {
+                  ..._1.data,
+                  commissioniApplicatePSP: "0.00"
+                },
+                elements: _1.elements
+              }),
+            commissioniApplicatePSP =>
+              right({
+                data: {
+                  ..._1.data,
+                  commissioniApplicatePSP
+                },
+                elements: _1.elements
+              })
+          )
+        )
+        .chain(_1 =>
+          getElementTextContentEither(
+            _.elements.datiPagamento,
+            "codiceEsitoPagamento"
+          ).map(codiceEsitoPagamento => ({
+            data: {
+              ..._1.data,
+              codiceEsitoPagamento
+            },
+            elements: _1.elements
+          }))
+        )
+    )
+    .chain(_ =>
+      DatiPagamento.decode(_.data).mapLeft(
+        err => new Error(err.map(_1 => _1.message).join("/"))
+      )
+    );
+}
+
+/**
+ * Parse indirizzoBeneficiario field value from an XML Document
+ */
+function parseIndirizzoBeneficiario(
+  xmlDocument: Document
+): Either<Error, IndirizzoBeneficiario> {
+  return fromOption(new Error("Missing field 'indirizzoBeneficiario'"))(
+    getElementFromXmlDocument(
+      xmlDocument,
+      "indirizzoBeneficiario"
+    ).mapNullable(e => e.textContent?.trim())
+  ).chain(indirizzoBeneficiario =>
+    IndirizzoBeneficiario.decode({ indirizzoBeneficiario }).mapLeft(
+      err => new Error(err.map(_ => _.message).join("/"))
+    )
+  );
+}
+
+/**
+ * Parse soggettoPagatore fields values from an XML Document
+ */
+function parseSoggettoPagatore(
+  xmlDocument: Document
+): Either<Error, SoggettoPagatore> {
+  return fromOption(new Error("Invalid soggettoPagatore"))(
+    getElementFromXmlDocument(xmlDocument, "soggettoPagatore").chain(
+      soggettoPagatore =>
+        getElementTextContent(soggettoPagatore, "anagraficaPagatore")
+          .map(anagraficaPagatore => ({
+            data: { anagraficaPagatore },
+            elements: { soggettoPagatore }
+          }))
+          .chain(_ =>
+            getElementTextContent(
+              _.elements.soggettoPagatore,
+              "codiceIdentificativoUnivoco"
+            ).map(codiceIdentificativoUnivoco => ({
+              data: {
+                ..._.data,
+                codiceIdentificativoUnivoco
+              },
+              elements: _.elements
+            }))
+          )
+          .chain(_ =>
+            getElementTextContent(
+              _.elements.soggettoPagatore,
+              "e-mailPagatore"
+            ).map(emailPagatore => ({
+              data: {
+                ..._.data,
+                emailPagatore
+              },
+              elements: _.elements
+            }))
+          )
+    )
+  ).chain(_ =>
+    SoggettoPagatore.decode(_.data).mapLeft(
+      err => new Error(err.map(_1 => _1.message).join("/"))
+    )
+  );
+}
+
+/**
+ * Parse a base64 encoded RT and store it inside a Blob Storage
```

Comment:
maybe it's worth to have two distinct methods (parse + store) to make tesing easier

Reply:
Parsing test could easily be performed mocking `blobStorageService` params provided to the handler. Could you suggest an alternative pattern?